src/node.rs

```diff
@@ -516,43 +535,51 @@ impl PbftNode {
                     Ok(msgs)
                 })?;
 
-        // Sync local state with actual state. We re-sync the sequence number with head
-        // again, even though we already did that in `on_block_new`, since if we're
-        // catching up as the primary, the sequence number should actually just be
-        // the head's block num, and not that+1.
-        if state.seq_num != head.block_num {
```

Comment:
This block of code was added in an earlier commit in this PR, so you should just fix it up with that commit to remove it altogether.